internal/conf/rate.go

```diff
@@ -0,0 +1,69 @@
+package conf
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Rate struct {
+	Events   float64       `json:"events,omitempty"`
+	OverTime time.Duration `json:"over_time,omitempty"`
+}
+
+func (r *Rate) EventsPerSecond() float64 {
+	if int64(r.OverTime) == 0 {
+		return r.Events
+	}
+
+	return r.Events / r.OverTime.Seconds()
+}
+
+// DefaultOverTime sets the OverTime field to overTime if it is 0.
+func (r *Rate) DefaultOverTime(overTime time.Duration) Rate {
+	if r.OverTime == 0 {
+		return Rate{
+			Events:   r.Events,
+			OverTime: time.Hour,
+		}
+	}
+
+	return *r
+}
+
+// Decode is used by envconfig to parse the env-config string to a Rate value.
+func (r *Rate) Decode(value string) error {
+	if f, err := strconv.ParseFloat(value, 64); err == nil {
+		r.Events = f
+		// r.OverTime remains 0 in this case
```

Comment:
maybe just set OverTime to `time.Hour` here? 
```suggestion
		r.OverTime = time.Hour
```

Reply:
Another good catch, would be a div by 0 @ EventsPerSecond.

Reply:
i think the point i was trying to make here was to handle setting the default for `OverTime` here rather than  catching the 0 division case in `EventsPerSecond()` - the original value for `OverTime` is meant to be an hour instead of 0 which doesn't seem to be handled currently?